frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -5905,6 +5837,21 @@ CallResolutionResult resolveTheseCall(ResolutionContext* rc,
   return resolveGeneratedCall(rc->context(), astContext, ci, inScopes);
 }
 
+const BuilderResult*
+buildDefaultFunction(Context* context,
+                     UniqueString typePath,
+                     UniqueString name) {
```

Comment:
IMO this function and all of the functions implementing it should be stored in `default-functions.cpp`. I know that is slightly out of pattern (since the prototype is in `resolution-queries.h`) but that can be addressed with a comment here or by moving these to a new public header (so that we have a public chpl/include/default-functions.h and a private one in lib/resolution/ with a different name).

Alternatively, you can keep `buildDefaultFunction` but move the various functions implementing it to `default-functions.cpp`.

Either way, I think it's important to move the helper functions implementing `buildDefaultFunction` to `default-functions.cpp` both for find-ability and because `resolution-queries.cpp` can get overly large because of it's grab-bag nature.

Reply:
I moved ``builderResultForDefaultFunction`` and ``buildTypeConstructor`` into default-functions.cpp.